tinygrad/renderer/wgsl.py

```diff
@@ -49,7 +49,8 @@ class WGSLRenderer(CStyleLanguage):
   code_for_op = {**CStyleLanguage.code_for_op, Ops.WHERE: lambda a,b,c,dtype: f"select({c},{b},{a})"}
   nan = "nan()"
   type_map = { dtypes.float: "f32", dtypes.uchar: "u32", dtypes.ushort: "u32", dtypes.short: "i32",
-              dtypes.char: "i32", dtypes.int32: "i32", dtypes.uint32: "u32", dtypes.bool: "bool" }
+              dtypes.char: "i32", dtypes.int32: "i32", dtypes.uint32: "u32", dtypes.bool: "bool",
+              dtypes.int64: "i32" } # wgsl has no int64, shapetracker upcast becomes a no-op
```

Comment:
i think we want to assert if it needs int64 and device does not support it. it's an illegal memory access that either fails silently or crashes